src/main/java/com/liferay/faces/util/application/ResourceValidatorFactory.java

```diff
@@ -26,8 +26,12 @@
 public abstract class ResourceValidatorFactory implements FacesWrapper<ResourceValidatorFactory> {
 
 	/**
-	 * @return  an instance of {@link ResourceValidator} from the {@link ResourceValidatorFactory} found by the {@link
-	 *          FactoryExtensionFinder}.
+	 * @return  a stateless, thread-safe (and potentially singleton) instance of {@link ResourceValidator} from the
```

Comment:
1. Show Neil the difference in generated javadoc of `@return` vs `Returns`.
2. Change `(and potentially singleton)` to  `singleton`.
3. Actually make objects returned by factories into Singleton. (in all jars)
4. Make sure that object returned by factories can be created at deploy time.
5. Test that everything runs in Tomcat and Wildfly.